### DISCUSSION TITLE
[Sass] Errors in basic use of Sass with 3rd party node modules

### DISCUSSION THREAD

**Discussion by btalb** [vercel/next.js#23945] [Help]
# Bug report

I have a SCSS file that compiles successfully manually but produces errors when using with NextJs. I can't understand what is being done differently to cause the issue.

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Create a new app with `npx create-next-app`
2. Install Sass `npm install sass`
3. Rename `global.css` to `global.scss`, & edit the import in `_app.js`
4. Install my 3rd party library that uses SCSS: `npm install @material/button`
5. Add a `next.config.js` with the config suggested by [the documentation](https://nextjs.org/docs/basic-features/built-in-css-support#customizing-sass-options) to include `node_modules` in Sass search path:
```javascript
const path = require('path')

module.exports = {
  sassOptions: {
    includePaths: [path.join(__dirname, 'node_modules')],
  },
}
```
6. Try to run / build the app with `npm run build`:
```
> test_next@0.1.0 build /home/user/tmp/test_next
> next build

Creating an optimized production build  

Failed to compile.

./styles/globals.scss
SassError: Can't find stylesheet to import.
   ╷
23 │ @use "@material/density/variables" as density-variables;
   │ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   ╵
  node_modules/@material/button/_variables.scss 23:1  @forward
  node_modules/@material/button/_index.scss 1:1       @use
  /home/ben/tmp/test_next/styles/globals.scss 1:1                                           root stylesheet


> Build error occurred
Error: > Build failed because of webpack errors
    at build (/home/user/tmp/test_next/node_modules/next/dist/build/index.js:15:918)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! test_next@0.1.0 build: `next build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the test_next@0.1.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/user/.npm/_logs/2020-08-06T22_18_48_929Z-debug.log
```
**Note: the "missing" stylesheets are definitely there**

## Expected behavior

Exactly the same as I get when I manually run Sass with the same configuration (i.e. no error):
```
user@pc:~/tmp/test_next$ npx sass -Inode_modules styles/globals.scss
html,
body {
  padding: 0;
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen, Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue, sans-serif;
}

a {
  color: inherit;
  text-decoration: none;
}

* {
  box-sizing: border-box;
}
```

## System information

- OS: Ubuntu 18.04
- Browser (if applies) Chrome 83.0.4103.61
- Version of Next.js: 9.5.1
- Version of Node.js: 12.16.3

## Additional context

Behaviour is really inconsistent (i.e. that `@use` statement works in my actual code, but another doesn't; whereas it didn't work in this minimal example), and I have no idea how to debug deeper than running a build / dev & seeing what spits out. Does NextJs have a document on debugging NextJs internals? How do I go about seeing what Webpack / Sass / loaders, etc. are actually doing under the hood?

Ideally I want to use this in CSS (SCSS) modules with components. Is this possible with NextJs, or will it be a hard slog?

**Comment by awareness481** [comment]
I could reproduce but it appears this was an issue with @material/button rather than Next.js.
Upon investigating the issue changing my `next.config.js` to this
```js
const path = require("path");

module.exports = {
  sassOptions: {
    includePaths: ["node_modules", "node_modules/@material/*"].map((d) =>
      path.join(__dirname, d)
    ),
  },
};
```

is what worked for me.

I also uploaded the [repo](https://github.com/awareness481/next-15961) if you prefer cloning.

**Comment by btalb** [comment]
Thanks @awareness481, that's helped progress things along for me.

Oddly, that fixes `@material/button` but not others (like `@material/tab-bar` for example).

I still believe there is something deeper going on here. To me it is odd that these examples work perfectly if I call Sass manually (`npx sass -Inode_modules ...`), & have come from a working Webpack system that wasn't using React or Next. 

I'm continuing the debugging train at the moment, but not getting too much insight. I've dug into `packages/next/build/webpack/config/blocks/css/index.ts` to try find what could be different between the systems that work & what Next is doing. I couldn't see anything that stood out, so I'm now onto dumping the `webpack` config in the `webpack` export from `next.config.js`. 

If you have any thoughts about where may be good to look, I'm all ears.

p.s. you said "it appears this was an issue with @material/button", do you have any insights into what made you think that?
